### PR TITLE
Add set_alarm_state, enable/disable_alarm_actions to v2 provider

### PR DESCRIPTION
**Pull request by baermat**
## Motivation
This ports the set_alarm_state and disable_alarm_actions functions to our new v2 provider

## Changes
- fix delete_alarms
- add set_alarm_state
- add disable_alarm_actions
- add enable_alarm_actions

## TODO

What's left to do:

- [x] set_alarm_state
- [x] disable_alarm_actions
- [x] enable_alarm_actions
- [x] check time formats

### CODE REVIEW

File: localstack/services/cloudwatch/provider_v2.py [hunk 1]
```diff
@@ -233,6 +245,63 @@ def get_metric_data(
...
+            actions = alarm.alarm["OKActions"]
+        elif state_value == "ALARM":
+            actions = alarm.alarm["AlarmActions"]
+        else:
+            actions = alarm.alarm["InsufficientDataActions"]
+        for action in actions:
+            data = arns.parse_arn(action)
+            # test for sns - can this be done in a more generic way?
+            if data["service"] == "sns":
+                service = connect_to.get_client(data["service"])
```
Comment (steffyP) [inline localstack/services/cloudwatch/provider_v2.py:293]: NIT: we probably should parse the arn first, and set the `region_name` and `aws_access_key_id` here as well, so we are on the safe side. 
Similar like we do it for the https://github.com/localstack/localstack/blob/master/localstack/services/s3/notifications.py#L543-L546

File: localstack/services/cloudwatch/provider_v2.py [hunk 2]
```diff
@@ -479,3 +548,102 @@ def get_metric_statistics(
...
+    ):
+        old_state = alarm.alarm["StateValue"]
+        store = self.get_store(context.account_id, context.region)
+        current_time = datetime.datetime.now()
+        store.Histories.append(
+            {
+                "Timestamp": timestamp_millis(alarm.alarm["StateUpdatedTimestamp"]),
+                "HistoryItemType": HistoryItemType.StateUpdate,
+                "AlarmName": alarm.alarm["AlarmName"],
+                "HistoryData": alarm.alarm.get("StateReasonData"),  # FIXME
```
Comment (steffyP) [inline localstack/services/cloudwatch/provider_v2.py:568]: Q: what needs to be fixed here? maybe you could add some more details to the note? 🙏

File: tests/aws/services/cloudwatch/test_cloudwatch.py [hunk 3]
```diff
@@ -1612,6 +1610,42 @@ def test_describe_minimal_metric_alarm(self, snapshot, aws_client, cleanups):
...
+        assert err["Code"] == "ValidationError"
+
+        with pytest.raises(Exception) as ex:
+            aws_client.cloudwatch.set_alarm_state(
+                AlarmName=f"{alarm_name}-nonexistent", StateValue="OK", StateReason="test"
+            )
+
+        err = ex.value.response["Error"]
+        assert err["Code"] == "ResourceNotFound"
+        assert not err.get("Message")
```
Comment (steffyP) [inline tests/aws/services/cloudwatch/test_cloudwatch.py:1647]: NIT: we could also snapshot the error-responses, e.g 
``` 
snapshot.match("error-resource-not-found", ex.value.response)
```

File: tests/aws/services/cloudwatch/test_cloudwatch.py [hunk 4]
```diff
@@ -1612,6 +1610,42 @@ def test_describe_minimal_metric_alarm(self, snapshot, aws_client, cleanups):
         response = aws_client.cloudwatch.describe_alarms(AlarmNames=[alarm_name])
         snapshot.match("describe_minimal_metric_alarm", response)
 
+    @markers.aws.validated
+    def test_set_alarm_invalid_input(self, aws_client, snapshot, cleanups):
```
Comment (steffyP) [inline tests/aws/services/cloudwatch/test_cloudwatch.py:1614]: Q: the test seems to fail with the old provider, do we need a `skipif` here?

**Review by steffyP — APPROVED** [review]
Nice work @baermat 👏 
Love to see the alarms working with the new provider 🚀 ⏰ 

I only added a few NITs, and I think we need to `skip` the new test for the old provider?

File: localstack/services/cloudwatch/provider_v2.py (hunk 1 above)
Comment (baermat) [reply localstack/services/cloudwatch/provider_v2.py:293]: :+1:

File: localstack/services/cloudwatch/provider_v2.py (hunk 2 above)
Comment (baermat) [reply localstack/services/cloudwatch/provider_v2.py:568]: We didn't really care for the history data so far and some of the content/formatting is off. Will clarify

File: tests/aws/services/cloudwatch/test_cloudwatch.py (hunk 3 above)
Comment (baermat) [reply tests/aws/services/cloudwatch/test_cloudwatch.py:1647]: :+1:

File: tests/aws/services/cloudwatch/test_cloudwatch.py (hunk 4 above)
Comment (baermat) [reply tests/aws/services/cloudwatch/test_cloudwatch.py:1614]: Thanks!

**Pull request merged**